File: identhub/src/main/java/de/solarisbank/identhub/session/data/contract/ContractSignNetworkDataSource.java
package de.solarisbank.identhub.session.data.contract;

import de.solarisbank.identhub.session.data.TransactionAuthenticationNumber;
import de.solarisbank.sdk.data.dto.IdentificationDto;
import io.reactivex.Single;
import okhttp3.ResponseBody;
import retrofit2.Response;

public interface ContractSignNetworkDataSource {
    Single<IdentificationDto> postAuthorize(final String identificationId);

    Single<IdentificationDto> postConfirm(final String identificationId, final TransactionAuthenticationNumber tan);

    Single<Response<ResponseBody>> fetchDocumentFile(final String documentId);
}

File: qes/src/main/java/de/solarisbank/identhub/qes/data/contract/ContractSignNetworkDataSource.kt
package de.solarisbank.identhub.data.contract

import de.solarisbank.sdk.data.dto.IdentificationDto
import io.reactivex.Single
import okhttp3.ResponseBody
import retrofit2.Response

interface ContractSignNetworkDataSource {

    fun postAuthorize(identificationId: String): Single<IdentificationDto>

    fun postConfirm(
        identificationId: String,
        tan: String
    ): Single<IdentificationDto>

    fun fetchDocumentFile(documentId: String): Single<Response<ResponseBody>>
}